src/Compilers/Core/Portable/FileSystem/FileUtilities.cs

```diff
@@ -223,7 +223,7 @@ internal static class FileUtilities
                 return null;
             }
 
-            string? normalizedPath = TryNormalizeAbsolutePath(resolvedPath);
+            string? normalizedPath = PathUtilities.IsAbsolute(resolvedPath) ? TryNormalizeAbsolutePath(resolvedPath) : null;
```

Comment:
Why did you put the check here vs. putting the check inside `TryNormalizeAbsolutePath`? Feels like one case it should handle is if it's not passed an absolute path.

Reply:
Reason I put it here was because it reduced the need to examine the other call sites in order to decide whether those call sites should "fail fast" when the input is not an absolute path, as they currently do, versus propagating a null result.

However, if it doesn't feel useful to preserve that "fail fast" behavior for the other call sites, I'm happy to push the check inside of `TryNormalizeAbsolutePath` and remove the debug assertion from within that method at the same time.